.github/workflows/regression_trusted.yml

```diff
@@ -0,0 +1,500 @@
+name: Regression Detector (trusted)
+
+on:
+  workflow_run:
+    workflows: ["Regression Detector"]
+    types:
+      - completed
+
+jobs:
+  compute-metadata:
+    name: Compute metadata for regression experiments
+    runs-on: ubuntu-22.04
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    outputs:
+      cpus: ${{ steps.system.outputs.CPUS }}
+      memory: ${{ steps.system.outputs.MEMORY }}
+
+      comparison-sha: ${{ steps.metadata.outputs.COMPARISON_SHA }}
+      comparison-tag: ${{ steps.metadata.outputs.COMPARISON_TAG }}
+      baseline-sha: ${{ steps.metadata.outputs.BASELINE_SHA }}
+      baseline-tag: ${{ steps.metadata.outputs.BASELINE_TAG }}
+      head-sha: ${{ steps.metadata.outputs.HEAD_SHA }}
+      checkout-sha: ${{ steps.metadata.outputs.CHECKOUT_SHA }}
+      github-event-number: ${{ steps.metadata.outputs.GITHUB_EVENT_NUMBER }}
+
+      replicas: ${{ steps.experimental-meta.outputs.REPLICAS }}
+      warmup-seconds: ${{ steps.experimental-meta.outputs.WARMUP_SECONDS }}
+      total-samples: ${{ steps.experimental-meta.outputs.TOTAL_SAMPLES }}
+      p-value: ${{ steps.experimental-meta.outputs.P_VALUE }}
+      smp-version: ${{ steps.experimental-meta.outputs.SMP_CRATE_VERSION }}
+
+    steps:
+      - name: Setup experimental metadata
+        id: experimental-meta
+        run: |
+          export WARMUP_SECONDS="45"
+          export REPLICAS="10"
+          export TOTAL_SAMPLES="600"
+          export P_VALUE="0.1"
+          export SMP_CRATE_VERSION="0.5.0"
+
+          echo "warmup seconds: ${WARMUP_SECONDS}"
+          echo "replicas: ${REPLICAS}"
+          echo "total samples: ${TOTAL_SAMPLES}"
+          echo "regression p-value: ${P_VALUE}"
+          echo "smp crate version: ${SMP_CRATE_VERSION}"
+
+          echo "WARMUP_SECONDS=${WARMUP_SECONDS}" >> $GITHUB_OUTPUT
+          echo "REPLICAS=${REPLICAS}" >> $GITHUB_OUTPUT
+          echo "TOTAL_SAMPLES=${TOTAL_SAMPLES}" >> $GITHUB_OUTPUT
+          echo "P_VALUE=${P_VALUE}" >> $GITHUB_OUTPUT
+          echo "SMP_CRATE_VERSION=${SMP_CRATE_VERSION}" >> $GITHUB_OUTPUT
+
+      - name: Setup system details
+        id: system
+        run: |
+          export CPUS="8"
+          export MEMORY="30g"
+
+          echo "cpus total: ${CPUS}"
+          echo "memory total: ${MEMORY}"
+
+          echo "CPUS=${CPUS}" >> $GITHUB_OUTPUT
+          echo "MEMORY=${MEMORY}" >> $GITHUB_OUTPUT
+
+      - name: 'Download metadata'
+        uses: actions/github-script@v3.1.0
```

Comment:
If memory serves, this version of `actions/github-script` is around 2 years old, and we ran into warnings re: this version on another project. That said, upgrading it was a pain, so keeping this version as-is to isolate the effects of the trusted/untrusted split seems like a safe path forward, if the Vector maintainers are in agreement. I can fish out a version of this script from another project when it's time to upgrade the `github-script` version; that script used version 6.2.0.

Reply:
Here's an approximate translation for `actions/github-script@v6.2.0`:

```javascript
            var artifacts = await github.paginate(github.rest.actions.listWorkflowRunArtifacts, {
               owner: context.repo.owner,
               repo: context.repo.repo,
               run_id: ${{github.event.workflow_run.id }},
            });

            var matchArtifact = artifacts.filter((artifact) => {
              return artifact.name == "meta"
            })[0];

            console.log("Downloading artifact %s", matchArtifact.id);

            var download = await github.actions.downloadArtifact({
               owner: context.repo.owner,
               repo: context.repo.repo,
               artifact_id: matchArtifact.id,
               archive_format: 'zip',
            });
            var fs = require('fs');
            fs.writeFileSync('${{github.workspace}}/meta.zip', Buffer.from(download.data));

```

Reply:
Edited above -- forgot to add a space between `${{` and `github.event.workflow_run.id }}`.

Reply:
Oh weird -- the version of this snippet in https://securitylab.github.com/research/github-actions-preventing-pwn-requests/ has the same quirk with `${{github.event.workflow_run.id }}`